### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by codepantha**
### Changes introduced via this PR 👇 
- [ ] [Install and setup devise](https://github.com/heartcombo/devise#getting-started).

- [ ] Can register a new user.

- [ ] User logs in with a combination of email and password.

- [ ] Hashed password should be stored in the database.

- [ ] Ask for confirmation of email.

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
+
```
Comment (Dipeshtwis) [inline app/views/devise/sessions/new.html.erb]: - [x] The views in devise for the registration and login should have been modified to match the wireframes and their styling.

### Your design 

### Sign in page 
Screenshot from 2022-03-02 03-04-04


### Sign-Up Page

Screenshot from 2022-03-02 03-04-15


### Wireframe design

Screenshot from 2022-03-02 03-09-41

File: app/views/shared/_user_head.html.erb [hunk 2]
```diff
@@ -7,5 +7,17 @@
       <h2 class="author"><%= @user.name %></h2>
     </a>
     <span>Number of posts: <%= @user.posts.count %></span>
+
+    
+  <%= form_with url: '/users/sign_out', method: :delete do |f| %>
+    <%= f.submit "Log out", class: 'submit' %>
+  <% end %>
+</div>
```
Comment (Dipeshtwis) [inline app/views/shared/_user_head.html.erb:15]: - [x] Looks like `Log out` is not working or at least after clicking the logout button, then also I'm able to get to that user's page. Please fix this.

Hint: you can use like this to logout the user session.
```
<%= link_to "Log out", destroy_user_session_path, method: :delete %>
```

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @codepantherr, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

:+1: Devise has been installed and setup.
:+1: I'm able to signup easily.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N.A.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/shared/_user_head.html.erb (hunk 2 above)
Comment (codepantha) [reply app/views/shared/_user_head.html.erb:15]: Hi @Dipeshtwis 👋  Thank you for your review. Please would you kindly explain what you mean by you can still get to the user's page? The user's page is not a resource that can only be accessed by authorized users - at least not yet; it's available for all users to check. The signout is working. If you are signed in then you can not visit the sign_in or sign_up page, but you can when you log out.

File: app/views/devise/sessions/new.html.erb (hunk 1 above)
Comment (codepantha) [reply app/views/devise/sessions/new.html.erb]: Thank you @Dipeshtwis. I will work on this 🙏

File: app/views/shared/_user_head.html.erb (hunk 2 above)
Comment (Dipeshtwis) [reply app/views/shared/_user_head.html.erb:15]: Hi @codepantherr,

Ohh!! That's my bad. Please fix other require changes. 

**Note for next reviewer**: Please do not count this as a require change. 

Thank you

Comment (codepantha) [reply app/views/shared/_user_head.html.erb:15]: Thank you @Dipeshtwis

File: app/views/devise/sessions/new.html.erb (hunk 1 above)
Comment (codepantha) [reply app/views/devise/sessions/new.html.erb]: This has been fixed.

**Review by DeliceLydia — APPROVED** [review]
# Congratulations! 🎉🎉

## Hi @codepantherr, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**